tests/src/bin/precompute_shares.rs

```diff
@@ -63,20 +63,28 @@ fn precompute_shares_for_curve<E: Curve, R: RngCore + CryptoRng>(
             .into_iter()
             .filter(|t| t.map(|t| t <= n).unwrap_or(true))
         {
-            eprintln!("t={t:?},n={n},curve={}", E::CURVE_NAME);
-            let primes = std::iter::repeat_with(|| {
-                let p = generate_blum_prime(rng, SecurityLevel128::SECURITY_BITS * 4);
-                let q = generate_blum_prime(rng, SecurityLevel128::SECURITY_BITS * 4);
-                (p, q)
-            })
-            .take(n.into())
-            .collect();
-            let shares = trusted_dealer::builder::<E, SecurityLevel128>(n)
-                .set_threshold(t)
-                .set_pregenerated_primes(primes)
-                .generate_shares(rng)
-                .context("generate shares")?;
-            cache.add_shares(t, n, &shares).context("add shares")?;
+            for hd_enabled in [false, true] {
+                eprintln!(
+                    "t={t:?},n={n},curve={},hd_enabled={hd_enabled}",
```

Comment:
Would be nice to have a test for loading keyshares between different crate features enabled. A bit hard to organize though.

Reply:
There are tests for that. On CI, we launch tests with and without hd support, so pregenerated key shares are loaded on both versions of the lib during the tests

Reply:
I don't see where this is done? Also, I meant having keyshares being generated with hd-wallet feature completely off as well

Reply:
We do have tests that generate keys, and we do run tests with hd feature disabled, see `test` and `test-hd` in the workflow:

https://github.com/dfns/cggmp21/blob/c003fa41e28a0ff9c3c2919839d7a5663a86f78c/.github/workflows/rust.yml#L28-L46

Reply:
Ah, it's the new workflow, I was looking in the master branch

Reply:
In the initial message I was also thinking of concerns that cargo sometimes still compiles with some features turned on. But let's leave that for later discovery

Reply:
You can see whether HD feature is on or off by looking at tests. If HD is on, there will be tests with HD suffix in their names